lib/suites/LinkedDataSignature.js

```diff
@@ -297,6 +307,19 @@ module.exports = class LinkedDataSignature {
         proof
       }));
 
+    // Check that the proofPurpose is valid
+    if(options.proofPurpose) {
+      const {expanded, proofPurpose} = options;
+      const verified = await proofPurpose.verify(
```

Comment:
Should we start employing the preferred validator convention with these new `verify` APIs?  That is, proofPurpsose.verify() can return `{valid: false, error: <error>}` instead of throwing?

Reply:
Yes, we should start employing the new API.